### DISCUSSION TITLE
Cannot Use File Upload in HTML

### DISCUSSION THREAD

**Discussion by KiddoV** [wailsapp/wails#2075] [Q&A]
I have a basic upload input in HTML:
```html
<input type="file" accept="image/png">
```
When I click on the ``Choose File`` button, nothing happen.
I tried putting the application on a browser using ``http://127.0.0.1:5173`` then it is working.
Am I missing something? Or Is this upload feature is disabled by default and I have to use ``runtime.OpenFileDialog()`` instead?
Thanks!

**Comment by leaanthony** [comment]
There's a few tickets/discussions about this so it looks like it's a hot topic. The Optimus project has DnD so I suggest you check that out to see if it does what you need. Otherwise feel free to comment on the existing tickets

**Reply by KiddoV** [reply]
Can you provide some links to this topic? Or reference the tickets? Thanks!

**Reply by KiddoV** [reply]
It works when I try it at home tho...hmmm
```console
Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       107.0.1418.35
npm             N/A             Installed       8.15.0
*upx            N/A             Installed       upx 3.96
*nsis           N/A             Available
```
Wonder if it is a webview2 problem or something else?

**Reply by leaanthony** [reply]
If you figure it out, let us know! Maybe we need to up the minimum version.

**Reply by KiddoV** [reply]
Will check the webview2 version on my work computer when I get back to work on Monday.

**Reply by KiddoV** [reply]
Hmm... so here is my webview2's version at work, it's even higher
```console
Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       107.0.1418.42
npm             N/A             Installed       8.15.0
*upx            N/A             Installed       upx 3.96
*nsis           N/A             Available
```

**Reply by KiddoV** [reply]
Again, it works in MSEdge browser, but not in Webview2 Runtime (Wails).

**Reply by stffabi** [reply]
That's very interesting, it's working here with your snippet on Windows 11 with the following Versions:

```
WebView2: 107.0.1418.42
Windows 11 - arm64: Version 10.0.22621.675
```

Edit: Also works on 107.0.1418.42

**Reply by KiddoV** [reply]
I don't know what to do know. Any suggestions?

**Comment by KiddoV** [comment]
This could fix the issue. Haven't try it yet.

https://github.com/MicrosoftEdge/WebView2Feedback/issues/2972

**Reply by leaanthony** [reply]
Thanks for letting us know 🙏 Looks like there still might be an issue? What would be your strategy for checking if the user installed runtime has support for this? I'm thinking that getting this info from the environment api call might be useful? Or maybe a capabilities API like `HTMLFileInput() bool` so you can change the frontend depending on capabilities?
